PokemonAPI/PokemonAPI.BusinessLayer/Implementations/DomainServices/BattleService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.EntityFrameworkCore;
+using PokemonAPI.BusinessLayer.Interfaces;
+using PokemonAPI.BusinessLayer.Models;
+using PokemonAPI.DomainLayer.Entities;
+using PokemonAPI.PersistenceLayer.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PokemonAPI.BusinessLayer.Implementations.DomainServices
+{
+    internal class BattleService : IBattleService
+    {
+
+        private readonly IRepository<Pokemon> _repository;
+
+        public BattleService(IRepository<Pokemon> repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task<BattleResult> Battle(Guid firstPokemonId, Guid secondPokemonId, CancellationToken cancellationToken = default)
+        {
+            var  firstPokemon = await _repository.Get(
+                filter: dbPokemon => dbPokemon.Id == firstPokemonId,
+                 include: source => source
+                    .Include(x => x.Abilities)
+                    .ThenInclude(x => x.Ability),
+                cancellationToken: cancellationToken
+            );
+
+            var secondPokemon = await _repository.Get(
+                filter: dbPokemon => dbPokemon.Id == secondPokemonId,
+                 include: source => source
+                    .Include(x => x.Abilities)
+                    .ThenInclude(x => x.Ability),
+                cancellationToken: cancellationToken
+            );
```

Comment:
DRY 
This is already implemented in the pokemon service and should be used as a source of truth. Instead of having a repository as a dependency here, we can reuse already existing implementation in the pokemon service which does it well. (It also handles null cases)